gammapy/spectrum/models.py

```diff
@@ -43,10 +148,16 @@ class PowerLaw(SpectralModel):
     reference : float, `~astropy.units.Quantity` 
         :math:`E_0`
     """
-    def __init__(self, index, amplitude, reference):
+    DEFAULT_INDEX = 2.3 * u.Unit('')
+    DEFAULT_AMPLITUDE = 2.5 * 1e-12 * u.Unit('cm-2 s-1 TeV-1')
```

Comment:
I'm still -1 on default model parameter values, but I won't insist on this.

The use case for wanting default model parameter values is convenience when prototyping.
That's not a very important use case that should decide about the implementation.
There are many other classes where you want to quickly create an object of that type when prototyping / testing, e.g. `EnergyDispersion`. There we also don't put a default unit matrix or Gaussian in `__init__`, but instead have alternative `from_*` constructors to conveniently and explicitly create such objects.

I'd prefer the same solution for models, and if those default models aren't useful to expose to end users, then just have a factory function or class in the tests that you can import and call (example: https://github.com/gammapy/gammapy/blob/1a2c88284af71f5573f0fdf9842184a85b2b600a/gammapy/image/tests/test_maps.py#L18)

My argument for removing the defaults would be that those are arbitrary values, and there's potential for confusion, maybe even mistakes by users. You also can't create a `SkyCoord` or `Time` or most other objects without passing data.

That said, some Astropy and Sherpa models do have defaults as well, but the values are usually 0 or 1, i.e. less magic numbers:
https://github.com/astropy/astropy/blob/master/astropy/modeling/powerlaws.py#L20
https://github.com/sherpa/sherpa/blob/master/sherpa/models/basic.py#L404

One other concern I had was whether the class-level and default argument implementation you have is possible to introduce weirdly connected model instances. Consider this:
```python
import astropy.units as u

class Model:
    par_default = [42, 43] * u.s
    def __init__(self, par=par_default):
        self.par = par

a = Model()
a.par.value[0] = 99
b = Model()
print(a.par)
print(a.par_default)
print(b.par)
print(b.par_default)
print(Model.par_default)
```
Now as far as I see, scalar quantity objects aren't mutable or at least hard to mutate?
So I don't think this implementation is going to bite users this way.